lib/new_relic/agent/llm/response_headers.rb

```diff
@@ -41,13 +50,30 @@ def populate_openai_response_headers(headers)
           self.request_id = headers[X_REQUEST_ID]&.first
           self.response_organization = headers[OPENAI_ORGANIZATION]&.first
           self.llm_version = headers[OPENAI_VERSION]&.first
-          self.rate_limit_requests = headers[X_RATELIMIT_LIMIT_REQUESTS]&.first
-          self.rate_limit_tokens = headers[X_RATELIMIT_LIMIT_TOKENS]&.first
-          self.rate_limit_remaining_requests = headers[X_RATELIMIT_REMAINING_REQUESTS]&.first
-          self.rate_limit_remaining_tokens = headers[X_RATELIMIT_REMAINING_TOKENS]&.first
+          self.rate_limit_requests = headers[X_RATELIMIT_LIMIT_REQUESTS]&.first&.to_i
+          self.rate_limit_tokens = headers[X_RATELIMIT_LIMIT_TOKENS]&.first&.to_i
+          remaining_headers(headers)
+          reset_headers(headers)
+          tokens_usage_based_headers(headers)
+        end
+
+        private
+
+        def remaining_headers(headers)
+          self.rate_limit_remaining_requests = headers[X_RATELIMIT_REMAINING_REQUESTS]&.first&.to_i
+          self.rate_limit_remaining_tokens = headers[X_RATELIMIT_REMAINING_TOKENS]&.first&.to_i
+        end
+
+        def reset_headers(headers)
           self.rate_limit_reset_requests = headers[X_RATELIMIT_RESET_REQUESTS]&.first
           self.rate_limit_reset_tokens = headers[X_RATELIMIT_RESET_TOKENS]&.first
         end
+
+        def tokens_usage_based_headers(headers)
+          self.rate_limit_limit_tokens_usage_based = headers[X_RATELIMIT_LIMIT_TOKENS_USAGE_BASED]&.first
```

Comment:
The value stays a string on this one?

Reply:
Ah, I see you both chatting about strings vs. ints elsewhere. It's odd to see given the variable name similarities, but that's how it has to be if the spec calls for it that way.

Reply:
Yes, it's part of the spec 🤷

I'm bummed that I haven't been able to create a request that returns this header so that I can find out what it actually looks like. I should probably check out what Python and Node are doing in their tests to update the example values. 

For some of the response headers, like `ratelimit_reset_requests`, the response is a time value, so it has a mix of numbers and letters: `4ms`. That might be true for this one as well.

Reply:
Whoops, missed your follow-up when I responded! Glad we're on the same page!